lib/DxilPIXPasses/PixPassHelpers.cpp

```diff
@@ -37,15 +37,33 @@ using namespace llvm;
 using namespace hlsl;
 
 namespace PIXPassHelpers {
-bool IsAllocateRayQueryInstruction(llvm::Value const *Val) {
-  if (Val != nullptr) {
-    if (llvm::Instruction const *Inst =
-            llvm::dyn_cast<llvm::Instruction>(Val)) {
-      return hlsl::OP::IsDxilOpFuncCallInst(Inst,
-                                            hlsl::OP::OpCode::AllocateRayQuery);
+static void FindRayQueryHandlesFromUse(Value *U,
+                                       std::vector<Value *> &Handles) {
+  Handles.push_back(U);
+  auto RayQueryHandleUses = U->uses();
+  for (Use &Use : RayQueryHandleUses) {
+    iterator_range<Value::user_iterator> Users = Use->users();
+    for (User *User : Users) {
+      if (std::find(Handles.begin(), Handles.end(), User) == Handles.end())
+        FindRayQueryHandlesFromUse(User, Handles);
```

Comment:
Not every user of anything that accepts a RayQuery handle on input is itself a ray query handle!  This should be careful only to follow users of specific instructions: PHI and Select for now.  No other instructions should have their users followed recursively.

Reply:
D'Oh! Of course... Thanks.